pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,288 @@
+        "OUTP:SEL?",
+        "OUTPUT:SEL %s",
+        "Set the selected channel to active or inactive or check its status.",
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    output_on = Instrument.control(
+        "OUTP:GEN?",
+        "OUTP:GEN %s",
+        "Set the output on or off or check the output status.",
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    # The following commands are for making it easier to change selected the
+    # selected channel and activate/deactivate them.
+
+    def activate_channel(self, channel):
+        """Activate a channel."""
+        # Save current selected channel before switching.
+        selected_channel = self.selected_channel
+        self.selected_channel = channel
+        self.selected_channel_active = True
+        # Restore previously selected channel.
+        self.selected_channel = selected_channel
+
+    def deactivate_channel(self, channel):
+        """Deactivate a channel."""
+        # Save current selected channel before switching.
+        selected_channel = self.selected_channel
+        self.selected_channel = channel
+        self.selected_channel_active = False
+        # Restore previously selected channel.
+        self.selected_channel = selected_channel
+
+    # Convenience functions to make turning on/off the output more natural.
+    def turn_output_on(self):
+        """Turn output on."""
+        self.output_on = True
+
+    def turn_output_off(self):
+        """Turn output off."""
+        self.output_on = False
+
+    # Measurement Commands ----------------------------------------------------
+
+    measured_voltage = Instrument.measurement(
+        "MEAS:VOLT?", "Measured voltage in V."
+    )
+
+    measured_current = Instrument.measurement(
+        "MEAS:CURR?", "Measured current in A."
+    )
+
+    # Arbitrary Sequence Commands ---------------------------------------------
+
+    def clear_sequence(self, channel):
+        """Clear the sequence of the selected channel."""
+        channel = strict_discrete_set(channel, [1, 2, 3, 4])
+        self.write(f"ARB:CLEAR {channel}")
+
+    sequence = Instrument.setting(
+        "ARB:DATA %s",
+        "Define sequence of triplets of voltage (V), current (A) and dwell "
+        "time (s).",
+        set_process=process_sequence,
+    )
+
+    repetitions = Instrument.control(
+        "ARB:REP?",
+        "ARB:REP %s",
+        "Number of repetitions (0...255). If 0 is entered, the sequence is"
+        "repeated infinitely.",
+        validator=strict_discrete_set,
+        values=range(256),
+        get_process=lambda x: int(x),
+    )
+
+    def load_sequence(self, slot):
+        """Load a saved waveform from internal memory (slot 1, 2 or 3)."""
+        slot = strict_discrete_set(slot, [1, 2, 3])
+        self.write(f"ARB:REST {slot}")
+
+    def save_sequence(self, slot):
+        """Save the sequence to internal memory (slot 1, 2 or 3)."""
```

Comment:
I guess this means the "current"/active sequence (that has been defined with `sequence`)? Should we point this out in the docstring?